### PR TITLE
Hugo final GSOC report.

### PR DESCRIPTION
**Pull request by hmacdope**
Hi all, my final GSOC report for posting on the blog.

### CODE REVIEW

File: _posts/2020-09-03-final-report-hugo.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    }
+
+}
+```
+Now you can just `make test` to ensure your TNG times will be fun times!
+
+Once we had a test suite that could be used to check library correctness on the
+major routines we started to work on the TNG library itself.
+Key goals were C++ compilation and in turn modernisation of the library
+routines to modern C++. I acheived full C++ compilation in a combination of [!26](https://gitlab.com/gromacs/tng/-/merge_requests/26), [!27](https://gitlab.com/gromacs/tng/-/merge_requests/27), [!29](https://gitlab.com/gromacs/tng/-/merge_requests/29), [!33](https://gitlab.com/gromacs/tng/-/merge_requests/33) and [!34](https://gitlab.com/gromacs/tng/-/merge_requests/34).
```
Comment (orbeckst) [inline _posts/2020-09-03-final-report-hugo.md:61]: achieved

(In any English ;-) )

File: _posts/2020-09-03-final-report-hugo.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+
+Refactoring the TNG library was a huge task, much more so than intially anticipated. Due to this and some
+changing circumstances, the aforementioned modernisation MRs are all still open with more work required. I
+plan on working on these into the future but progress will be slow as I do not
+have solid blocks of time to dedicate. 
+
+All the work I have done on TNG itself can be found in my TNG
+[fork](https://gitlab.com/hugomacdermott/tng) as well as open PRs on the
+library [itself](https://gitlab.com/gromacs/tng).
```
Comment (orbeckst) [inline _posts/2020-09-03-final-report-hugo.md:118]: I would make the link target more explicit
```suggestion
[fork](https://gitlab.com/hugomacdermott/tng) as well as open PRs on the
[TNG library itself](https://gitlab.com/gromacs/tng).
```

File: _posts/2020-09-03-final-report-hugo.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+
+# PyTNG time!
+
+From here I switched focus to [PyTNG](https://github.com/MDAnalysis/pytng), a
+set of python bindings designed for use by MDAnalysis although technically a
```
Comment (orbeckst) [inline _posts/2020-09-03-final-report-hugo.md:124]:
```suggestion
set of Python bindings designed for use by MDAnalysis although technically a
```

File: _posts/2020-09-03-final-report-hugo.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+separate library. I worked on PyTNG for most of the second half of GSOC. This required
+changing gears a little bit as well as learning Cython, which was initially a
```
Comment (orbeckst) [inline _posts/2020-09-03-final-report-hugo.md:126]: Maybe link Cython?

File: _posts/2020-09-03-final-report-hugo.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+Firstly I changed the TNG libraries exported with PyTNG itself to
+include the bugfixes obtained as part of the earlier improvements to the TNG
+library. This was incorporated as part of
+[#28](https://github.com/MDAnalysis/pytng/pull/28). I then improved the TNG library
+calls in PyTNG in [#29](https://github.com/MDAnalysis/pytng/pull/29).
+
+Following some design discussions, we then moved towards a newer design for the
+bindings, so as to be able to read all the blocks available in a TNG file with
+maximal speed. This was acheived in merging
```
Comment (orbeckst) [inline _posts/2020-09-03-final-report-hugo.md:137]:
```suggestion
maximal speed. This was achieved in merging
```

File: _posts/2020-09-03-final-report-hugo.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+[#32](https://github.com/MDAnalysis/pytng/pull/32) and ongoing work in
```
Comment (orbeckst) [inline _posts/2020-09-03-final-report-hugo.md:138]: If you want to re-use links then you can do this markdown with
```
[#32][PR32] and [#38][PR38], which are total


[#32]: https://github.com/MDAnalysis/pytng/pull/32
[#38]: https://github.com/MDAnalysis/pytng/pull/38
```
See https://github.github.com/gfm/#reference-link

File: _posts/2020-09-03-final-report-hugo.md [hunk 7]
```diff
@@ -0,0 +1,213 @@
+---
+layout: post
+title:  "GsoC Report: Trajectory New Generation: the trajectory format for the future of simulation"
```
Comment (orbeckst) [inline _posts/2020-09-03-final-report-hugo.md:3]: Correct capitalization
```suggestion
title:  "GSoC Report: Trajectory New Generation: the trajectory format for the future of simulation"
```

File: _posts/2020-09-03-final-report-hugo.md [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+date:   2020-09-03 08:58:08 +1000
+categories: 
+---
+
+# This is the end?
+
+Well well well, here we are at the end of GSOC! What a ride it has been, giving
```
Comment (orbeckst) [inline _posts/2020-09-03-final-report-hugo.md:10]:
```suggestion
Well well well, here we are at the end of GSoC! What a ride it has been, giving
```

File: _posts/2020-09-03-final-report-hugo.md [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+
+GSOC was a fantastic experience for me, as I felt very welcome amongst two
+communities (MDAnalysis and GROMACS). I hope I was able to give back! I was
+constantly challenged throughout and this helped me learn and grow with big
+improvements both in my technical skills and approach to computational problem
+solving.
+
+*Thanks all for having me along!*
+
+@hmacdope
```
Comment (orbeckst) [inline _posts/2020-09-03-final-report-hugo.md:211]: By convention
```suggestion
— @hmacdope 
```

**Review by orbeckst — APPROVED** [review]
Very nice, I enjoyed reading your post, @hmacdope . I only have minor comments/typos.

I built the site locally and it renders fine.

File: _posts/2020-09-03-final-report-hugo.md (hunk 8 above)
Comment (orbeckst) [reply _posts/2020-09-03-final-report-hugo.md:10]: Should be consistent throughout... so if you go with GSoC then change the other GSOC -> GSoC occurencesm, too.

**Comment by hmacdope** [comment]
Thanks @orbeckst, spelling never my strong suit ...

**Comment by hmacdope** [comment]
Ready to go from my end. :)

**Pull request merged**